File: frontend-web/types/notification.ts
export type NotificationType = 1 | 2 | 3 | 4;
export type NotificationTargetType = 1 | 2 | 3;

export type NotificationItem = {
  id: number;
  actorId: number;
  actorUsername?: string;
  actorAvatar?: string;
  type: NotificationType;
  targetType: NotificationTargetType;
  targetId: number;
  content?: string | null;
  isRead: number;
  createTime: string;
};

File: src/main/java/teektok/entity/Notification.java
package teektok.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("notification")
public class Notification {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private Long actorId;
    private Integer type;
    private Integer targetType;
    private Long targetId;
    private String content;
    private Integer isRead;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    private LocalDateTime createTime;
}
